pygmt/src/grdtrack.py

```diff
@@ -292,29 +289,30 @@ def grdtrack(grid, points=None, newcolname=None, outfile=None, **kwargs):
     if hasattr(points, "columns") and newcolname is None:
         raise GMTInvalidInput("Please pass in a str to 'newcolname'")
 
-    with GMTTempFile(suffix=".csv") as tmpfile:
-        with Session() as lib:
-            with lib.virtualfile_from_data(
-                check_kind="raster", data=grid
-            ) as grdfile, lib.virtualfile_from_data(
-                check_kind="vector", data=points, required_data=False
-            ) as csvfile:
-                kwargs["G"] = grdfile
-                if outfile is None:  # Output to tmpfile if outfile is not set
-                    outfile = tmpfile.name
-                lib.call_module(
-                    module="grdtrack",
-                    args=build_arg_string(kwargs, infile=csvfile, outfile=outfile),
-                )
+    with Session() as lib:
+        with lib.virtualfile_from_data(
+            check_kind="raster", data=grid
+        ) as grdfile, lib.virtualfile_from_data(
+            check_kind="vector", data=points, required_data=False
+        ) as csvfile, lib.virtualfile_to_gmtdataset() as outvfile:
+            kwargs["G"] = grdfile
+            lib.call_module(
+                module="grdtrack",
+                args=build_arg_string(kwargs, infile=csvfile, outfile=outvfile),
+            )
+        if outfile is not None:
+            # if output_type == "file":
```

Comment:
For grdtrack and other similar modules, we need to add the `output_type` parameter.